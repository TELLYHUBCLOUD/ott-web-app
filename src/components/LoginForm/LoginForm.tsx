import React from 'react';
import type { LoginFormData } from 'types/account';
import type { FormErrors } from 'types/form';

import Button from '../Button/Button';
import { getLoginUrl } from '../../services/sso.service';

import styles from './LoginForm.module.scss';

type Props = {
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  error?: string;
  errors: FormErrors<LoginFormData>;
  values: LoginFormData;
  submitting: boolean;
  siteName?: string;
};

const LoginForm: React.FC<Props> = () => {
  return (
    <div className={styles.wrapper}>
      <span>Your session has expired, please login again.</span>
      <Button color="primary" label={'Sign in Again'} to={getLoginUrl()} />
    </div>
  );
};

export default LoginForm;
