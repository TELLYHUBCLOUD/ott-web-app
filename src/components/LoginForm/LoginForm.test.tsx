import React from 'react';

import { render } from '../../testUtils';

import LoginForm from './LoginForm';

describe('<LoginForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <LoginForm
        onSubmit={jest.fn()}
        onChange={jest.fn()}
        values={{
          email: '',
          password: '',
        }}
        errors={{}}
        submitting={false}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
