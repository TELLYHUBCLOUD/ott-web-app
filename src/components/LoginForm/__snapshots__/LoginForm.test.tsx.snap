// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm> renders and matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <span>
      Your session has expired, please login again.
    </span>
    <a
      class="button primary outlined"
      href="#"
    >
      <span>
        Sign in Again
      </span>
    </a>
  </div>
</div>
`;
